skycoin/skywire
# Connect to services over dmsghttp

Did you run `make format && make check`?
yes

Fixes #985 

 Changes:	
- Updated `disc.NewHTTP`
- Added `http.Client` param to `dmsgc.New()`
- Update `initDmsg()` to connect to the `dmsg-discovery` via HTTP over dmsg if the URL of `dmsg-discovery` in the config is `dmsg://<pk-of-dmsg-discovery>:80`
- Added `dmshhttp` module
- Renamed `localServers.json` to `dmsghttp-config.json`
- Changed datatype of `DMSGServers` from `[]string` to `[]*disc.Entry`

How to test this PR:

On test deployment:
1. Run
    ```
    ./skywire-cli config gen -dirt
    ```
2. Run 
    ```
    ./skywire-visor skywire-config.json
    ```
3. Open hypervisor UI and check functionality. 

On integration env:
Check [#skywire-services-486](https://github.com/SkycoinPro/skywire-services/pull/486)

Needs:
[#dmsg-130](https://github.com/skycoin/dmsg/pull/130)
